core/src/main/java/com/yahoo/oak/InternalOakMap.java

```diff
@@ -980,25 +1154,28 @@ private OakScopedReadBuffer getNextChunkMinKey(OrderedChunk<K, V> c) {
          */
         @Override
         protected boolean advanceState() {
-            boolean valueToReturn = super.advanceState();
-
-            if (valueToReturn) {
-                BasicChunk<K, V> chunk = getState().getChunk();
-                BasicChunk.BasicChunkIter chunkIter = getState().getChunkIter();
-                int nextIndex = getState().getIndex();
-                // The boundary check is costly and need to be performed only when required,
-                // meaning not on the full scan.
-                if (((OrderedChunk<K, V>.ChunkIter) chunkIter).isBoundCheckNeeded()) {
-                    chunk.readKeyFromEntryIndex(ctx.tempKey, nextIndex);
-                    if (!inBounds(ctx.tempKey)) {
-                        setState(null);
-                        valueToReturn = false;
+            while (true) {
```

Comment:
why do we need this "while(true)"? apparently, it does nothing, the return statement is always activated in the first loop.